src/Raven.Client/Documents/Session/AbstractDocumentQuery.cs

```diff
@@ -932,7 +932,57 @@ public void AndAlso()
 
             tokens.AddLast(QueryOperatorToken.And);
         }
+        
+        /// <summary>
+        ///   Wraps previous query with clauses and adds an AND operator to the given query
+        /// </summary>
+        public void AndAlso(bool wrapPreviousQueryClauses)
+        {
+            var tokens = GetCurrentWhereTokens();
+            
+            if (tokens.Last == null)
+                return;
 
+            if (tokens.Last.Value is QueryOperatorToken)
+                throw new InvalidOperationException("Cannot add AND, previous token was already an operator token.");
+            
+            if (wrapPreviousQueryClauses == false)
+            {
+                tokens.AddLast(QueryOperatorToken.And);
+            }
+            else
+            { 
+                tokens.AddFirst(OpenSubclauseToken.Create());
+                tokens.AddLast(CloseSubclauseToken.Create());
+                tokens.AddLast(QueryOperatorToken.And);
+            }
+        }
+
+        /// <summary>
+        ///   Wraps previous query with clauses and adds an OR operator to the given query
+        /// </summary>
+        public void OrElse(bool wrapPreviousQueryClauses)
+        {
+            var tokens = GetCurrentWhereTokens();
+
+            if (tokens.Last == null)
+                return;
+
+            if (tokens.Last.Value is QueryOperatorToken)
+                throw new InvalidOperationException("Cannot add OR, previous token was already an operator token.");
+            
+            if (wrapPreviousQueryClauses == false)
+            {
+                tokens.AddLast(QueryOperatorToken.Or);
+            }
+            else
+            { 
+                tokens.AddFirst(OpenSubclauseToken.Create());
+                tokens.AddLast(CloseSubclauseToken.Create());
+                tokens.AddLast(QueryOperatorToken.Or);
+            }
+        }
+        
```

Comment:
I don't think we need this

Reply:
From binary logic point of view probably not, but wondering if there is a reason not to have it. Sometimes people organize their intent by wrapping some binary blocks into subclauses.

Reply:
It is more about how likely this is to cause people to do the wrong thing.

Reply:
Let's remove it then, I do not insist to have it.